sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala

```diff
@@ -166,7 +166,14 @@ private[parquet] object ParquetFilters {
    * Converts data sources filters to Parquet filter predicates.
    */
   def createFilter(schema: StructType, predicate: sources.Filter): Option[FilterPredicate] = {
-    val dataTypeOf = getFieldMap(schema)
+    val nameTypeMap = getFieldMap(schema)
+
+    // Parquet does not allow dots in the column name because dots are used as a column path
```

Comment:
Instead of checking dots for each predicate recursively in `ParquetFilters`, we can check `Filter.references` of the predicate at top level in `ParquetFileFormat`,  and skip `ParquetFilters.createFilter` at all.